Source/pyboy/botsupport/tilemap.py

```diff
@@ -0,0 +1,193 @@
+
+    @property
+    def signed_tile_index(self):
+        """
+        The Game Boy uses both signed and unsigned tile indexes. Use this attribute to identify which is used. Read more about it in [Pan Docs: VRAM Tile Data](http://bgb.bircd.org/pandocs.htm#vramtiledata).
+
+        Returns:
+            bool: Whether this tile map uses signed tile indexes.
+        """
+        return self.signed_tile_data
+
+    def get_tile_address(self, x, y):
+        """
+        Returns the memory address in the tilemap for the tile at the given coordinate. The address contains the index of tile which will be shown at this position. This should not be confused with the actual tile data of `pyboy.botsupport.tile.Tile.data_address`.
+
+        This can be used as an global identifier for the specific location in a tile map.
+
+        Be aware, that the tile index referenced at the memory address might change between calls to `pyboy.pyboy.PyBoy.tick`. And the tile data for the same tile index might also change to display something else on the screen.
+
+        The index might also be a signed number. Depending on if it is signed or not, will change where the tile data is read from. Use `pyboy.botsupport.tilemap.TileMap.signed_tile_index` to test if the indexes are signed for this tile view. You can read how the indexes work in the [Pan Docs: VRAM Tile Data](http://bgb.bircd.org/pandocs.htm#vramtiledata).
+
+        Args:
+            x (int): X-coordinate in this tile map.
+            y (int): Y-coordinate in this tile map.
+
+        Returns:
+            int: Address in the tile map to read a tile index.
+        """
+
+        if not 0 <= x < 32:
+            raise IndexError("x is out of bounds. Value of 0 to 31 is allowed")
+        if not 0 <= y < 32:
+            raise IndexError("y is out of bounds. Value of 0 to 31 is allowed")
+        return self.map_offset + 32*y + x
+
+    def get_tile(self, x, y):
+        """
+        Provides a `pyboy.botsupport.tile.Tile`-object which allows for easy interpretation of the tile data. The object is agnostic to where it was found in the tilemap. I.e. equal `pyboy.botsupport.tile.Tile`-objects might be returned from two different coordinates in the tile map.
+
+        Args:
+            x (int): X-coordinate in this tile map.
+            y (int): Y-coordinate in this tile map.
+
+        Returns:
+            `pyboy.botsupport.tile.Tile`: Tile object corresponding to the tile index at the given coordinate in the tile map.
+        """
+        return Tile(self.mb, index=(self.get_tile_index(x,y)[1], self.signed_tile_data))
+
+    def get_tile_identifier(self, x, y):
+        """
+        Returns an identifier of the tile at the given coordinate in the tile map. The identifier can be used to quickly recognize what is on the screen through this tile view.
+
+        This identifier unifies the otherwise complicated indexing system on the Game Boy into a single range from 0-384.
```

Comment:
```suggestion
        This identifier unifies the otherwise complicated indexing system on the Game Boy into a single range of [0, 384).
```

Reply:
Although mathematically correct, I'm not sure our audience understands it right away. I for one had to look it up to be sure.

Reply:
Not that mine is clear about it either

Reply:
Now, I have written `... 0-383 (both included)`